File: src/main/java/io/openfuture/openmessanger/service/UserAuthService.java
package io.openfuture.openmessanger.service;

import javax.validation.constraints.NotNull;

import com.amazonaws.services.cognitoidp.model.AdminListUserAuthEventsResult;
import com.amazonaws.services.cognitoidp.model.ForgotPasswordResult;

import io.openfuture.openmessanger.service.dto.LoginRequest;
import io.openfuture.openmessanger.service.dto.LoginSmsVerifyRequest;
import io.openfuture.openmessanger.service.dto.RefreshTokenRequest;
import io.openfuture.openmessanger.service.dto.UserPasswordUpdateRequest;
import io.openfuture.openmessanger.service.dto.UserSignUpRequest;
import io.openfuture.openmessanger.service.response.UserResponse;
import io.openfuture.openmessanger.web.response.AuthenticatedResponse;


public interface UserAuthService {

    AuthenticatedResponse authenticate(LoginRequest userLogin);

    AuthenticatedResponse refreshToken(RefreshTokenRequest request);

    AuthenticatedResponse authenticateSms(LoginSmsVerifyRequest loginSmsVerifyRequest);

    AuthenticatedResponse updateUserPassword(UserPasswordUpdateRequest userPasswordUpdateRequest);

    void logout(@NotNull String accessToken);

    ForgotPasswordResult userForgotPassword(String username);

    void createUser(UserSignUpRequest signUpDTO);

    AdminListUserAuthEventsResult userAuthEvents(String username, int maxResult, String nextToken);

    UserResponse getCurrent();
}

File: src/main/kotlin/io/openfuture/openmessanger/service/UserAuthService.kt
package io.openfuture.openmessanger.service

import com.amazonaws.services.cognitoidp.model.AdminListUserAuthEventsResult
import com.amazonaws.services.cognitoidp.model.ForgotPasswordResult
import io.openfuture.openmessanger.service.dto.*
import io.openfuture.openmessanger.service.response.UserResponse
import io.openfuture.openmessanger.web.response.AuthenticatedResponse
import javax.validation.constraints.NotNull

interface UserAuthService {
    fun authenticate(userLogin: LoginRequest): AuthenticatedResponse
    fun refreshToken(request: RefreshTokenRequest): AuthenticatedResponse
    fun authenticateSms(loginSmsVerifyRequest: LoginSmsVerifyRequest): AuthenticatedResponse
    fun logout(accessToken: @NotNull String?)
    fun userForgotPassword(username: String?): ForgotPasswordResult?
    fun createUser(signUpDTO: UserSignUpRequest)
    fun userAuthEvents(username: String?, maxResult: Int, nextToken: String?): AdminListUserAuthEventsResult?
    fun current(): UserResponse
}